apps/web/components/Breadcrumbs.tsx

```diff
@@ -0,0 +1,82 @@
+"use client";
+import { usePathname } from "next/navigation";
+import Link from "next/link";
+import React, { useEffect, useMemo, useState } from "react";
+import { ChevronRightIcon } from "@heroicons/react/24/solid";
+import { getTitlesFromUrlSegments } from "@/services/getTitlesFromUrlSegments";
+
+interface Breadcrumb {
+  href: string;
+  label: string;
+}
+
+const truncateString = (str: string) => {
+  return `${str.slice(0, 6)}...${str.slice(-4)}`;
+};
+
+export function Breadcrumbs() {
+  const path = usePathname();
+  const [breadcrumbs, setBreadcrumbs] = useState<Breadcrumb[]>([]);
+
+  const breadcrumbsPromise = useMemo(async (): Promise<Breadcrumb[]> => {
```

Comment:
Not sure that useMemo is saving computing in that case, this is only saving the function declaration witch is nothing but the code inside is not executed until called so no need toi memoize it

Reply:
you can go with simple function declaration

Reply:
And using useMemo is costly so we shouldnt use it if it doesnt bring significant performance saving

Reply:
got it!